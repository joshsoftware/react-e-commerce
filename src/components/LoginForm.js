import React from 'react';
import { Form } from 'reactstrap';
import ButtonWrapper from './ButtonWrapper';
import FormEmailField from './FormEmailField';
import FormPasswordField from './FormPasswordField';
import PropTypes from 'prop-types';

const LoginForm = ({ validateData, dispatch, formState }) => {
  const { email, password, emailError, passwordError } = formState;

  return (
    <>
      <h3>Login</h3>
      <hr />
      <Form>
        <FormEmailField email={email} emailError={emailError} dispatch={dispatch} />
        <FormPasswordField password={password} passwordError={passwordError} dispatch={dispatch} />
        <ButtonWrapper buttonText={'Login'} onClick={validateData} />
        <hr />
        <ButtonWrapper buttonText={'Sign In With Google'} />
      </Form>
    </>
  );
};

export default LoginForm;

LoginForm.propTypes = {
  validateData: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  formState: PropTypes.object.isRequired
};
