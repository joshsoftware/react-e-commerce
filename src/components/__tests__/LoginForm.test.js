import { render, cleanup } from '@testing-library/react';
import React from 'react';
import LoginForm from '../LoginForm';

afterEach(cleanup);

describe('LoginForm Component', () => {
  const validateData = () =>{

  }
  const dispatch = () => {

  }
  const formState = {
    email: 'abc',
    password: 'ada'
  }
  it('Must return LoginForm Component', () => {
    const { asFragment } = render(<LoginForm validateData={validateData} dispatch={dispatch} formState={formState}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
