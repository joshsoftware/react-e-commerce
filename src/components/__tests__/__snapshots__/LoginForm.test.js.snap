// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component Must return LoginForm Component 1`] = `
<DocumentFragment>
  <h3>
    Login
  </h3>
  <hr />
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleEmail"
      >
        Email
      </label>
      <input
        aria-invalid="true"
        class="is-invalid form-control"
        name="email"
        placeholder="example@company.com"
        type="email"
        value="abc"
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="examplePassword"
      >
        Password
      </label>
      <input
        aria-invalid="true"
        class="is-invalid form-control"
        name="password"
        placeholder="password"
        type="password"
        value="ada"
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <button
      class="btn btn-secondary"
      type="button"
    >
      Login
    </button>
    <hr />
    <button
      class="btn btn-secondary"
    >
      Sign In With Google
    </button>
  </form>
</DocumentFragment>
`;
